tools/raisinbread_refresh.php

```diff
@@ -0,0 +1,218 @@
+ * password with the Raisinbread data set is not used.
+ *
+ * In order to prevent accidental data loss, the script prompts the user to
+ * manually type the exact name of the database which will be affected.
+ *
+ * In normal cases, the database connection will be established via the 
+ * credentials in the config file. However, if the database tables have already
+ * been deleted then this will not work. In this case it is still possible to
+ * import Raisinbread data if the user has properly set up a MySQL configuration
+ * file and provides the name of their LORIS database.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+
+$info = <<<INFO
+This script is used by LORIS developers to DELETE DATA and replace
+them with new test data.
+
+
+INFO;
+
+echo $info;
+
+echo "\e[0;31;40m*** WARNING this will result in the LOSS OF DATA ***\e[0m\n";
+
+$cwd = getcwd();
+if (substr_compare($cwd, 'tools', mb_strlen($cwd) - mb_strlen('tools')) !== 0) {
+    die('Please run this script from the tools/ directory.' . PHP_EOL);
+}
+
+try {
+    require_once 'generic_includes.php';
+
+    $dbInfo = $config->getSettingFromXML('database');
+
+    if (! $config->getSetting('dev')['sandbox']) {
+        throw new \LorisException(
+            "Config file indicates that this is not a sandbox. Aborting to " .
+            "prevent accidental data loss."
+        );
+    }
+    $dbname = $dbInfo['database'];
+    $host = $dbInfo['host'];
+    $username = $dbInfo['username'];
+    $password = $dbInfo['password'];
+
+    $urlConfigSetting = $config->getSetting('url');
+    $baseConfigSetting = $config->getSetting('base');
+    $hostConfigSetting = $config->getSetting('host');
+
+} catch (\DatabaseException $e) {
+    echo 'Could not connect to the database in the Config file.' .
+        'It\'s possible that the LORIS tables have already been dropped.' .
+        PHP_EOL;
+    echo 'Continue attempting to install Raisinbread database? (yN)' . PHP_EOL;
+    $input = trim(fgets(STDIN));
+
+    if (mb_strtolower($input) !== 'y') {
+        die;
+    }
+    echo PHP_EOL . 'Please enter the name of your database:' . PHP_EOL;
+    $dbname = trim(fgets(STDIN));
+} catch (\LorisException $e) {
+    die($e->getMessage());
+} catch (Exception $e) {
+    die("Could not load project/config.xml");
+}
+
+
+echo <<<CONFIRMATION
+Please type the database name `$dbname` to confirm you wish to drop tables
+and import test data: 
+CONFIRMATION;
+
+$input = trim(fgets(STDIN));
+if ($input !== $dbname) {
+    die('Input did not match database name. Exiting.');
+}
+
+// Create a connection to mysql via bash. All credentials are supplied via 
+// command line arguments except for the password which must be entered manually.
+$mysqlCommand = "mysql -A \"$dbname\"";
```

Comment:
here, it would be nice so that the mysql user, host and password could be read from the `quatuser` of the config.xml file instead of having to set `.my.conf` for it to work

Reply:
It's possible to connect via the config file instead already. I was using the main user instead of the quatUser though. I'll update this.